### DISCUSSION TITLE
Pagination and Connection

### DISCUSSION THREAD

**Discussion by lambrosx77** [Netflix/dgs-framework#576] [Q&A]
Hi,

It should be interesting if we can have the opportunity to add an optional **totalCount** field to Connection like that:

```
type MessageConnection {
    totalCount: Int
    edges: [MessageEdge]
    pageInfo: PageInfo
}

type MessageEdge {
    node: Message
    cursor: String
}

type PageInfo {
    hasPreviousPage: Boolean!
    hasNextPage: Boolean!
    startCursor: String
    endCursor: String
}
```

```
@Data
public class RelayConnection<T extends Object> extends DefaultConnection<T> {
    private int totalCount;

    public RelayConnection(int totalCount, List<Edge<T>> edges, PageInfo pageInfo) {
        super(edges, pageInfo);
        this.totalCount = totalCount;
    }
}
```

Thank you.

**Comment by srinivasankavitha** [comment]
@lambrosx77 -  the generated connection types are mainly for the simple use case where you just want the fields per the relay spec. For adding additional fields, we are recommending that the user just define them manually. Not sure how easy it would be to support what you describe, but happy to hear your thoughts/ideas around this.

**Reply by lambrosx77** [reply]
@srinivasankavitha Thank you for your anwser.

Common use case of pagination is paginated table like [here](https://react-table.tanstack.com/docs/api/usePagination) and we need totalCount to determine the number of pages. 
Graphql suggests totalCount in the [preview](https://graphql.org/learn/pagination/#end-of-list-counts-and-connections) of pagination too.

So, I agree with you, if we need specific, we have to implement it by hand but if it's a common use case and easy to integrate, it could save a lot of time to teams. 

By the way, thank you for the good work and saving time to community.

**Reply by yamalameyooooo** [reply]
@lambrosx77 did you implemented the totalCount? if yes how to, i am using codegen to run the project. SO, type MessageConnection will be generated in runtime, how to add a new field totalCount to it and how to write its implementation ? 
@srinivasankavitha maybe u can help me too in this if possible

**Reply by srinivasankavitha** [reply]
If you want additional fields, you will need to avoid the use of `@connection` to generate paginated types and just add them by hand.

**Reply by yamalameyooooo** [reply]
you mean i have to define 
```
edges(cursor, node)
pageInfo(hasPreviousPage, hasNextPage, startCursor, endCursor)
totalCount 
```
in my schema file and write only totalCounts logic or i have to implement the logic for pagination too? @srinivasankavitha

**Reply by yamalameyooooo** [reply]
any example can i have for this particular use case, as i am using codegen to generate files using schema @srinivasankavitha

**Comment by srinivasankavitha** [comment]
Yes, you have to explicitly define the paginated types in your schema and
also implement it. This is because you want to customize the paginated
types. Codegen does not do anything special for supporting pagination, but
if you have them defined in your schema, you will get the corresponding
generated types for those. We generally follow the relay spec as specified
here: https://relay.dev/graphql/connections.htm

Hope that helps.
